src/Root.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Root of the React component tree.
+ * @flow
+ */
+
+var React = require('react'),
+    Controls = require('./Controls'),
+    GenomeTrack = require('./GenomeTrack'),
+    types = require('./types'),
+    // TODO: make this an "import type" when react-tools 0.13.0 is out.
+    TwoBit = require('./TwoBit');
+
+var Root = React.createClass({
+  propTypes: {
+    referenceSource: React.PropTypes.object.isRequired
+  },
+  getInitialState: function(): any {
+    return {
+      contigList: ['1', '2', '3'],
+      range: null,
+      basePairs: null
+    }
+  },
+  componentDidMount: function() {
+    // Note: flow is unable to infer this type through `this.propTypes`.
+    var ref: TwoBit = this.props.referenceSource;
+    ref.getContigList().then(contigList => {
+      this.setState({contigList});
+    });
+  },
+  handleRangeChange: function(newRange: GenomeRange) {
```

Comment:
`types.GenomeRange` is the `React.PropTypes` version of this type. The Flow version of the type is in `types/types.js`. I was unable to namespace the types in that file.

This is all around gross. See https://github.com/facebook/flow/issues/267.